src/ejabberd_router.erl

```diff
@@ -268,9 +277,9 @@ check_component_global(LDomain, Node) ->
     %% check that there is no component registered globally for this node
     case get_global_component(LDomain, Node) of
         undefined ->
-            ok;
+            false;
         _ ->
-            false
+            true
```

Comment:
Ok, so we had a hidden bug, maybe we need to add a test?

Reply:
Not a bug, it was working correctly. But I think that having a validation function that returns `boolean() | ok`, when only `ok`value indicates a successful check, is super confusing. So I unified and reworked it a bit.